### ISSUE TITLE
lint: add checker that detects suspicious re-assignments

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#401]
Detects suspicious re-assignments.

**Before:**
```go
xs[0] = i+0
xs[1] = i+1
xs[0] = i+2 // Potentially incorrect index on the LHS
```
**After:**
```go
xs[0] = i+0
xs[1] = i+1
xs[2] = i+2 // Correct index
```
